pkg/kubernetes/artefact_downloader.go

```diff
@@ -0,0 +1,128 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/log"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	kubernetesDir = "kubernetes"
+
+	rke2ReleaseURL = "https://github.com/rancher/rke2/releases/download/%s/%s"
+
+	rke2Binary     = "rke2.linux-%s.tar.gz"
+	rke2CoreImages = "rke2-images-core.linux-%s.tar.zst"
+	rke2Checksums  = "sha256sum-%s.txt"
+
+	rke2CalicoImages = "rke2-images-calico.linux-%s.tar.zst"
+	rke2CanalImages  = "rke2-images-canal.linux-%s.tar.zst"
+	rke2CiliumImages = "rke2-images-cilium.linux-%s.tar.zst"
+	rke2MultusImages = "rke2-images-multus.linux-%s.tar.zst"
+
+	rke2VSphereImages = "rke2-images-vsphere.linux-%s.tar.zst"
```

Comment:
k3s is somewhat similar but because it doesn't have the option for different CNI's it's just one image (`k3s-airgap-images-amd64.tar.zst`) for the container images.